src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -340,6 +342,18 @@ def render_definition_template(
         project_context[CONTEXT_KEY]["env"] = environment_overrides
         return ProjectProperties(project_definition, project_context)
 
+    has_user_referenced_vars = (
```

Comment:
can we put this whole new logic into a function?

Reply:
I also refactored the usage [here](https://github.com/snowflakedb/snowflake-cli/pull/1597/commits/c41044e2be6d1a1fdf8bc5acd1672aaee590c0f4#diff-c52e43f6818ff4f456b3184aa96e4e2f23e295252adc90d4be2fa7cd65bda89fR337) with this function, let me know if this works for you!